lib/compile/formats.js

```diff
@@ -33,7 +33,8 @@ formats.fast = {
   // email (sources from jsen validator):
   // http://stackoverflow.com/questions/201323/using-a-regular-expression-to-validate-an-email-address#answer-8829363
   // http://www.w3.org/TR/html5/forms.html#valid-e-mail-address (search for 'willful violation')
```

Comment:
So I've done some more research here.
Turns out that hostnames can only contain letters/digits/hyphens.
However domain names can have mare characters
See this stackoverflow post with some references to RFCs: http://stackoverflow.com/a/2183140

So if I'm understanding this correctly, this would be valid: "foo@my_sub.domain.com"
But this wouldn't be: "foo@my_domain.com"

The email RFC says that the domain part is either a hostname or a dns domain name or an IP in square brackets (it basically refers to those RFCs)